packages/api/src/module/write/learning-materials-tasks/domain/complete-task.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { CompleteTask } from '@/module/commands/complete-task.domain-command';
+import { TaskWasCompleted } from '@/module/events/task-was-completed.domain-event';
+
+import { completeTask } from './complete-task';
+
+describe('complete task', () => {
+  let command: CompleteTask;
+
+  it('should return task was completed', () => {
+    // Given
+    const pastEvents: TaskWasCompleted[] = [];
+
+    // When
+    const events = completeTask(pastEvents, command);
+
+    // Then
+    expect(events).toStrictEqual([
+      {
+        type: 'TaskWasCompleted',
+        data: { learningMaterialsId: command.data.learningMaterialsId, taskId: command.data.taskId },
+      },
+    ]);
+  });
+
+  it('should throw exception if task was already completed', () => {
+    // Given
+    const pastEvents: TaskWasCompleted[] = [
+      {
+        type: 'TaskWasCompleted',
+        data: { learningMaterialsId: command.data.learningMaterialsId, taskId: command.data.taskId },
+      },
+    ];
+
+    // When
+    const events = () => completeTask(pastEvents, command);
+
+    // Then
+    expect(events).toThrowError('Task was already completed');
+  });
+
+  function setup() {
```

Comment:
It's a little strange usage of setup for me. 
More readable would be use just function which returns command in place of command field.